go/tools/builders/generate_test_main.go

```diff
@@ -166,9 +168,17 @@ func coverRegisterFile(fileName string, counter []uint32, pos []uint32, numStmts
 {{end}}
 
 func main() {
+{{if .AltRunDir}}
```

Comment:
I wonder if instead of baking it into the binary through the template, we should just add a standard flag to the test main, and set it by default in the go_test.
That way if you build the test binary and then invoke it by hand the default behavior is not surprising (few tests actually look at the file system, and yet you would require the runfiles directory to be able to run them by hand)